src/runtime/internal/await_block.ts

```diff
@@ -59,6 +59,9 @@ export function handle_promise(promise, info) {
 			update(info.then, 1, info.value, value);
 			set_current_component(null);
 		}, error => {
+			if (!info.hasCatch) {
```

Comment:
I don't think we want to throw this error before the code below. We don't want to break Svelte's internal state if there's a rejected promise in an `{#await}` with no `{:catch}`. We just want there to be an uncaught promise rejection that you can listen for.

Reply:
hmm... throwing in a callback function of `then` would lead to uncaught promise rejection right?

also, if there's no `info.hasCatch`, then also dont need to update the `info.catch` block right?

Reply:
@Conduitry, Oh I didn't think about it. Should I move this if block after `set_current_component(null)`?

Reply:
Okay, Looking at it, I did think about it. I was thinking like

 if there is no catch component, updating it is not necessary at all right?. it still seems to work either way, which one is better? (As I am not too much familiar with Svelte internals)

Reply:
I saw the `set_current_component()` call, and missing one of those would almost certainly break things - but I didn't notice that here we're skipping both setting it and then unsetting it, which I guess should be fine.

Reply:
Okay, so this should be fine as it is now right?